iotdb-core/confignode/src/main/java/org/apache/iotdb/confignode/manager/partition/PartitionManager.java

```diff
@@ -1448,6 +1464,54 @@ public void stopRegionCleaner() {
     }
   }
 
+  /**
+   * The ConfigNode-leader will periodically invoke this interface to automatically clean expired
+   * partition table.
+   */
+  public void autoCleanDataPartitionTable() {
+    List<String> databases = getClusterSchemaManager().getDatabaseNames(null);
+    Map<String, Long> databaseTTLMap = getClusterSchemaManager().getTTLInfoForUpgrading();
+    for (String database : databases) {
+      long subTreeMaxTTL = getTTLManager().getDatabaseMaxTTL(database);
+      databaseTTLMap.put(
+          database, Math.max(subTreeMaxTTL, databaseTTLMap.getOrDefault(database, -1L)));
```

Comment:
add some judgement like "isDatabaseExisted(database) && 0 < ttl && ttl < Long.MAX_VALUE" here to remove overhead? 
BTW, If all the databases don't have ttl, we can logically just do this check and find that none of them need to be cleaned up, so there's no need to do a consensus write

Reply:
Thanks for pinpointing this logic enhancement. The judgement is available at PartitionTableAutoCleaner.